packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -193,13 +193,25 @@ export function createObjectProxy<TSchema extends FlexObjectNodeSchema>(
 
 			return true;
 		},
+		// The "has" trap is called when a user checks for a property with "in" (e.g. `if ("z" in point) {...}`).
 		has: (target, key) => {
-			return (
+			if (
 				schema.objectNodeFields.has(key as FieldKey) ||
 				(allowAdditionalProperties ? Reflect.has(target, key) : false)
-			);
+			) {
+				// Filter out properties which are not own-enumerable, for example, the keys of optional fields that are in the schema, but whose value is currently `undefined`.
```

Comment:
"in" is supposed to return true for non-enumerable properties, as well as inherited properties, so this logic doesn't seem correct. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in

I think you want:

if (schema.objectNodeFields.has(key as FieldKey) {
return fieldIsNotEmpty
} else {
 return allowAdditionalProperties && Reflect.has(target, key)
}

Reply:
Agreed.